### PR TITLE
feat(rest): connect use url and token

### PR DESCRIPTION
**Pull request by dingbo8128**

### CODE REVIEW

**Comment by dingbo8128** [comment]
Main changes:
 1. remove parameter host and port, replace them with parameter "url"
 2. add example and modify test cases
 3. update readme

File: taos/cinterface.py [hunk 1]
```diff
@@ -266,7 +267,9 @@ def taos_connect_auth(host=None, user="root", auth="", db=None, port=0):
     connection = c_void_p(_libtaos.taos_connect_auth(_host, _user, _auth, _db, _port))
 
     if connection.value is None:
-        raise ConnectionError("connect to TDengine failed")
+        errno = taos_errno()
```
Comment (zitsen) [inline taos/cinterface.py:270]: Why there's no null_ptr?

**Review by zitsen — APPROVED** [review]
LGTM. see my comments.

**Comment by zitsen** [comment]
> Main changes:
> 
> 1. remove parameter host and port, replace them with parameter "url"
> 2. add example and modify test cases
> 3. update readme

Better write it in a git commit message body.

File: README.md [hunk 2]
```diff
@@ -31,10 +31,9 @@ pip install git+https://github.com/taosdata/taos-connector-python.git
 import taosrest
 
 # all parameters  are optional
-conn = taosrest.connect(host="localhost",
+conn = taosrest.connect(url="http://localhost:6041",
```
Comment (sangshuduo) [inline README.md:34]: we need to modify the official doc to reflect the change too.

**Review by sangshuduo — COMMENTED** [review]
see my comments

**Pull request merged**